python/paddle/v2/framework/tests/test_dynamic_recurrent_op.py

```diff
@@ -107,5 +111,58 @@ def test_forward(self):
         print 'pd_output', pd_output
 
 
+class RecurrentGradientOpTest(unittest.TestCase):
+    py = PyRNNStep()
+
+    def create_forward_op(self):
+        # create RNNOp
+        self.forward_op = DynamicRecurrentOp(
+            # inputs
+            inlinks=["x"],
+            boot_memories=["h_boot"],
+            step_net="stepnet",
+            # outputs
+            outlinks=["h@mem"],
+            step_scopes="step_scopes",
+            # attributes
+            pre_memories=["h@pre"],
+            memories=["h@mem"])
+
+    def create_gradient_op(self):
+        a = set()
+        backward_op = core.DynamicRecurrentOp.backward(self.forward_op, a)
+
+    def create_step_net(self):
```

Comment:
In the design, we use Python's `with` to construct the step block. It seems that this Python test should also do that.

Reply:
This is an underlying API, the with-statement syntax is not supported yet because the python API is still under developing.